nni/algorithms/compression/v2/pytorch/base/pruner_tools.py

```diff
@@ -0,0 +1,390 @@
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT license.
+
+import logging
+import types
+from typing import List, Dict, Optional, Callable, Union
+
+import torch
+from torch import Tensor
+from torch.nn import Module
+from torch.optim import Optimizer
+
+from .compressor import Compressor, LayerInfo
+
+_logger = logging.getLogger(__name__)
+
+__all__ = ['DataCollector', 'TrainerBasedDataCollector', 'HookCollectorInfo', 'MetricsCalculator', 'SparsityAllocator']
+
+
+class DataCollector:
```

Comment:
This is because the pruner base class uses these three tools. And puts the base class in `common` is also reasonable, we can discuss this in the next meeting.